packages/shared/lib/services/config.service.ts

```diff
@@ -236,6 +259,38 @@ class ConfigService {
 
         return { copiedToId: providerConfigResponse.id!, copiedFromId: foundConfigId };
     }
+
+    VALIDATION_RULES: ValidationRule[] = [
+        {
+            field: 'oauth_client_id',
+            modes: ['OAUTH1', 'OAUTH2', 'TBA', 'APP', 'CUSTOM'],
+            isValid: (config) => !!config.oauth_client_id
+        },
+        {
+            field: 'oauth_client_secret',
+            modes: ['OAUTH1', 'OAUTH2', 'TBA', 'APP', 'CUSTOM'],
+            isValid: (config) => !!config.oauth_client_secret
+        },
+        {
+            field: 'app_link',
+            modes: ['APP', 'CUSTOM'],
+            isValid: (config) => !!config.app_link
+        },
+        {
+            field: 'app_id',
+            modes: ['CUSTOM'],
+            isValid: (config) => !!config.custom?.['app_id']
+        },
+        {
+            field: 'private_key',
+            modes: ['CUSTOM'],
+            isValid: (config) => !!config.custom?.['private_key']
+        }
+    ];
+
+    validateProviderConfig(authMode: AuthModeType, providerConfig: ProviderConfig): string[] {
+        return this.VALIDATION_RULES.flatMap((rule) => (rule.modes.includes(authMode) && !rule.isValid(providerConfig) ? [rule.field] : []));
+    }
```

Comment:
not for change: Could have been Zod imo, not necessarily quicker but more flexible
```ts
const val = z.discriminatedUnion('type', [
   z.object({ 
     type: z.enum('OAUTH1', 'OAUTH2'),
     oauth_client_id: z.string(), 
     oauth_client_secret: z.string() 
   })
  ]
)
val.safeParse({ type: authMode, ...providerConfig });
```

Reply:
oh yeah using `zod` for validation is also a good idea